aya/src/programs/links.rs

```diff
@@ -369,6 +369,180 @@ pub enum LinkError {
     SyscallError(#[from] SyscallError),
 }
 
+/// A [`Link`] identifier, which may or may not be owned by aya.
+pub struct LinkId(u32);
+
+impl LinkId {
+    /// A wrapper for an arbitrary loaded link specified by its kernel id,
+    /// unsafe because there is no guarantee provided by aya that the link is
+    /// still loaded or even exists.
+    pub unsafe fn new(id: u32) -> Self {
+        Self(id)
+    }
+}
+
+bitflags::bitflags! {
+    /// Flags which are use to build a set of MprogOptions.
+    #[derive(Clone, Copy, Debug, Default)]
+    pub(crate) struct MprogFlags: u32 {
+        const REPLACE = BPF_F_REPLACE;
+        const BEFORE = BPF_F_BEFORE;
+        const AFTER = BPF_F_AFTER;
+        const ID = BPF_F_ID;
+        const FD = BPF_F_LINK;
+    }
+}
+
+/// Struct defining the arguments required for interacting with the kernel's
+/// multi-prog API.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is 6.6.
+///
+/// # Example
+///
+///```no_run
+/// # let mut bpf = aya::Ebpf::load(&[])?;
+/// use aya::programs::{tc, SchedClassifier, TcAttachType, tc::TcAttachOptions, LinkOrder};
+///
+/// let prog: &mut SchedClassifier = bpf.program_mut("redirect_ingress").unwrap().try_into()?;
+/// prog.load()?;
+/// let options = TcAttachOptions::tcxoptions(LinkOrder::first());
+/// prog.attach_with_options("eth0", TcAttachType::Ingress, options)?;
+///
+/// # Ok::<(), aya::EbpfError>(())
+/// ```
+#[derive(Debug)]
+pub struct LinkOrder {
```

Comment:
Can you add a "new" constructor so that LinkOrder can be created directly without calling one of the functions?